crates/turborepo-ffi/src/lockfile.rs

```diff
@@ -67,3 +96,10 @@ fn npm_subgraph_inner(buf: Buffer) -> Result<Vec<u8>, Error> {
     let contents = real_npm_subgraph(&request.contents, &request.workspaces, &request.packages)?;
     Ok(contents)
 }
+
+impl proto::PackageDependency {
+    pub fn into_tuple(self) -> (String, String) {
```

Comment:
Can we name this tuple with either a type or struct instead of leaving it anonymous?

Reply:
I _think_ you're using a tuple because it's easier to pass over the wire?

Reply:
This definitely needs a comment, but my initial thought it that the ordering of tuples is stable, while the rust compiler is free to reorder structs which could cause memory issues across different compiler versions (or even compiler invocations..).

Since we don't own proto::PackageDependency, we can't apply the `#[repr(C)]` attribute.

Reply:
I was just using this so I could get a map from `list.into_iter().map(PackageDependency::into_tuple).collect()`. Moved this logic to a `From` implementation to make the intention clearer.